File: 20A40/cir-agus/EJER_34/Empleado.cpp
# include <iostream>
# include <string>
# include "Empleado.h"

Empleado::Empleado(){}

Empleado::Empleado(std::string newNombre,double newSalario,std::string newFecha){
    this -> nombre = newNombre;
    this -> salario = newSalario;
    this -> fecha = newFecha;
}

bool Empleado::setNombre(std::string nom){
    if(nom.length() > 0){
        nombre = nom;
        return true;
    } else return false;    
}

bool Empleado::setSalario(double sal){
    if(sal > 0){
        salario = sal;
        return true;
    }else return false;
}

bool Empleado::setFecha(std::string fecha){
    if(fecha.length() > 0){
        fecha = fecha;
        return true;
    } else return false;
}

std::string Empleado::getNombre(){ return nombre; }
double Empleado::getSalario(){ return salario; }
std::string Empleado::getFecha() {return fecha;}

std::string Empleado::toString(){
    std::string cadena = "Nombre: "+nombre;
    cadena += "Salario: "+std::to_string(salario);
    cadena += "Fecha: "+fecha;
    return cadena;
}

File: 20A40/cir-agus/EJER_36/Empleado.py
class Empleado:
    def __init__(self,Nombre, Salario, fechaNac) -> None:
        self.__nombre = Nombre
        self.__salario = Salario
        self.__fechaNac = fechaNac
        
    def setNombre(self,nomb) -> bool:
        if len(nomb) > 0: 
            self.__nombre = nomb   
            return True
        else: return False
    
    def setSalario(self,sala) -> bool:
        if sala > 0:    
            self.__salario = sala
            return True
        else: return False
    
    def setFechaNac(self, fech) -> bool:
        if len(fech) > 0:
            self.__fechaNac = fech
            return True
        else: return False

    def getNombre(self) -> str: return self.__nombre
    def getSalario(self) -> float: return self.__salario
    def getFechaNac(self) -> str: return self.__fechaNac
    
    def toString(self) -> str:
        resultado = f"Nombre: {self.__nombre}\n"
        resultado += f"Salario: {self.__salario}\n"
        resultado += f"Fecha nacimiento: {self.__fechaNac}"
        return resultado